app/controllers/application_controller.rb

```diff
@@ -3,30 +3,56 @@
 class ApplicationController < ActionController::Base
   include GDS::SSO::ControllerMethods
   before_action :set_request_id
-  before_action :authenticate_user!
-  before_action :set_user_instance_variable
+  before_action :authenticate
   before_action :check_service_unavailable
   default_form_builder GOVUKDesignSystemFormBuilder::FormBuilder
 
   before_action :clear_questions_session_data
 
-  def check_service_unavailable
-    if ENV["SERVICE_UNAVAILABLE"].present?
-      render "errors/service_unavailable", status: :service_unavailable, formats: :html
+  def authenticate
+    if Settings.basic_auth.enabled
+      basic_auth
+    else
+      signon_auth
     end
   end
 
-  def set_user_instance_variable
+  def basic_auth
+    request.env["warden"].manager.config.intercept_401 = false
+    authenticate_or_request_with_http_basic do |name, password|
+      name == Settings.basic_auth.username && password == Settings.basic_auth.password
+    end
```

Comment:
Any reason we're not using `http_basic_authenticate_or_request_with` here?

https://rubydoc.info/github/rails/rails/ActionController%2FHttpAuthentication%2FBasic%2FControllerMethods:http_basic_authenticate_or_request_with

```suggestion
    http_basic_authenticate_or_request_with (
      name: Settings.basic_auth.username,
      password: Settings.basic_auth.password,
    )
```

Reply:
No, think that should work